### ISSUE TITLE
test_rasterize_dpi fails with 1.5.1

### ISSUE THREAD

**Issue by felixonmars** [matplotlib/matplotlib#5829]
I am getting only one test failure for both Python 2.7.11 and 3.5.1 on Arch. Hope this helps :)

```
======================================================================
FAIL: matplotlib.tests.test_image.test_rasterize_dpi.test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/python-matplotlib/src/matplotlib-1.5.1-py2/build/lib.linux-x86_64-2.7/matplotlib/testing/decorators.py", line 53, in failer
    result = f(*args, **kwargs)
  File "/build/python-matplotlib/src/matplotlib-1.5.1-py2/build/lib.linux-x86_64-2.7/matplotlib/testing/decorators.py", line 220, in do_test
    '(RMS %(rms).3f)'%err)
ImageComparisonFailure: images not close: /build/python-matplotlib/src/tmp_test_dir/result_images/test_image/rasterize_10dpi_svg.png vs. /build/python-matplotlib/src/tmp_test_dir/result_images/test_image/rasterize_10dpi-expected_svg.png (RMS 5.661)

----------------------------------------------------------------------
Ran 5165 tests in 136.270s

FAILED (KNOWNFAIL=18, SKIP=1, failures=1)
Shape of x does not match that of z: found (9, 9) instead of (9, 10).
```

**Comment by tacaswell** [comment]
hmm, I saw the same thing locally in https://github.com/matplotlib/matplotlib/pull/5817 (I use arch :smile: ) but everything is passing on travis so I wrote it off as a local issue....

**Comment by jenshnielsen** [comment]
I'm seeing it too. Not sure why Travis doesn't catch it

**Comment by mdboom** [comment]
Just as another data point -- I *can't* reproduce in Python 2.7.11 or 3.5.1 (Conda on Fedora).  Can someone who's hitting this post the SVG file so I can see *how* it's failing?

**Comment by mdboom** [comment]
I'm able to reproduce it now -- I think I wasn't getting a full rebuild earlier.

**Comment by tacaswell** [comment]
Sorry everyone :disappointed: .

**Comment by mdboom** [comment]
I think what we're running into is how Inkscape is handling image edge rendering -- which appears to be different in the version of Inkscape on Travis (which I think is 0.48.1), and that on our local machines (which is probably the latest 0.91).  The old Inkscape will resample the image such that it "bleeds" outside of the image bounding box.  This creates the same rendering before and after the matplotlib change.  The new Inkscape seems to strictly crop the image to the image bounding box, creating a different rendering between the two images -- the old has bleed (since it wasn't cropped correctly) and the new one doesn't.  

The bottom line is that I think it's safe to update the image to the latest generated in this case.  There isn't actually a matplotlib bug here as far as I can tell -- the entire image content (and nothing more) is making it into the output SVG, which is exactly right.

**Comment by tacaswell** [comment]
Probably need to add a test on inkscape version too so we can skip on travis?

**Comment by mdboom** [comment]
> Probably need to add a test on inkscape version too so we can skip on travis?

Hopefully by updating the baseline image, we won't be Inkscape-version-dependent anymore.
